core/vm/instructions_zkevm.go

```diff
@@ -225,14 +225,6 @@ func makeLog_zkevm(size int, logIndexPerTx bool) executionFunc {
 				}
 
 				d, _ = hex.DecodeString(dataHex)
-			} else {
-				// erigon behaviour
```

Comment:
Changeset looks good, however I don't think we can remove this - there is a case where the bug is not possible, above the 'bug trigger blockheight' where we will not correctly populate 'd'?

Reply:
But in this case, the log data does not need padding zeros in forkId.9. So I am a little confused
```
# This is a zkevm node rpc result with forkID.9
curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0x6118d61806013e84408a47bd1bad1e44149534bc2f4b4dae5f440fe11618c579"],"id":1}' https://rpc.xlayer.tech | jq
```

Reply:
What about zkevm mainnet?

Reply:
Got it, I will test zkevm mainnet

Reply:
I have tested on zkevm mainnet with branch **okx:zjg/fix-block-hash**, It got the same block hash form zkevm node.
So I think we can delete [zkEvm] fill 0 at the end?
```
# erigon zjg/fix-block-hash rpc node
 curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["13105691",true],"id":1}' http://127.0.0.1:8545/
```
```
{"jsonrpc":"2.0","id":1,"result":{"difficulty":"0x0","extraData":"0x","gasLimit":"0x4000000000000","gasUsed":"0x5208","hash":"0x68a36749226cfcb9e50611c2069921b5372e3559d80d34aba5208d4ce4333dae","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","miner":"0x148ee7daf16574cd020afa34cc658f8f3fbd2800","mixHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000","number":"0xc7fa1b","parentHash":"0x3be47dbf1ac29ffae5822801ed3bc0068995360bf6d62832f42b23017e597fa7","receiptsRoot":"0x056b23fbba480696b65fe5a59b8f2148a1299103c4f57df839233af2cf4ca2d2","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","size":"0x276","stateRoot":"0x0dee161078a0adfe25fce3d398f20f7f00bba92585d9fe621db003003f55c488","timestamp":"0x6660643e","totalDifficulty":"0x0","transactions":[{"blockHash":"0x68a36749226cfcb9e50611c2069921b5372e3559d80d34aba5208d4ce4333dae","blockNumber":"0xc7fa1b","from":"0x9af3049dd15616fd627a35563b5282bea5c32e20","gas":"0x5208","gasPrice":"0x472afe80","hash":"0x9e3d21eb0d1636bad495ca235b827882fa6b7c2bc4c4d7ff37e23e9be5b4da39","input":"0x","nonce":"0x70870","to":"0x417a7ba2d8d0060ae6c54fd098590db854b9c1d5","transactionIndex":"0x0","value":"0x9184e72a000","type":"0x0","chainId":"0x44d","v":"0x8bd","r":"0xeb935366231fc74c3c5289bac8c78e7987c1d6c64e2b66ba64795bba893cca7f","s":"0x7b4eecea7d7ba472f45a6a65b28c2173f9eab74466ae23bcd57a06e1192a430e"}],"transactionsRoot":"0x6fc12a5a2b9c9668291270175d56df0d61d902a088c8b2e9c0ae154bdde8c092","uncles":[]}}
```

Reply:
Cool, we probably need to make sure that it also works on Bali & Cardona before we can merge…

Reply:
But overall looks all right to me.

Reply:
> Cool, we probably need to make sure that it also works on Bali & Cardona before we can merge…

Okay, I will test on Bali & Cardona. Maybe Bali use protobuffer ds, I need create an new branch first to test on Bali.